### DISCUSSION TITLE
Be able to use custom conversion method using that maps to existing object in sub mapping

### DISCUSSION THREAD

**Discussion by PaulVrugt** [riok/mapperly#658] [Ideas]
I've been experimenting with mapperly, and we seriously consider dropping automapper an migrating (in steps) to mapperly. All kinds of stuff that I though wouldn't work in my test project actually already works out of the box. My compliments for this. This is a very cool library.

One thing that doesn't seem the work is the following regarding User implemented mapping methods:

Say I have a class:

```c#
public class SourceClass
{
    public SubSourceClass SubEntity {get; set; }
}

public class TargetClass
{
    public required TargetSubClass SubEntity {get; set; } = new();
}
```

and in my mapper I have a converter that maps to an existing `TargetClass`, and a custom converter for `SubSourceClass`:
```c#
//mapping method to existing TargetClass
public partial void ToTargetClass(SourceClass source, TargetClass target);
//custom mapping function to existing TargetSubClass
public static void ToTargetSubClass(SubSourceClass source, TargetSubClass target) {}
```

It would be really cool to be able to have mapperly recognize that when generating the mapping, it can use the `ToTargetSubClass` method, since I'm mapping to an existing `TargetClass` and the `SubEntity` property already has a value (either by checking the value, or in this case even because the `SubEntity` property is a non-nullable required property

**Comment by latonz** [comment]
With 3.1.0 this should work if the `TargetClass.SubEntity` property is not settable. It does not work for required settable properties, as Mapperly thinks it can just map the value to a new instance and assign that.
